diff --git a/build/configs/istax_multi.mk b/build/configs/istax_multi.mk
index fbaff9c..fd587a7 100644
--- a/build/configs/istax_multi.mk
+++ b/build/configs/istax_multi.mk
@@ -54,6 +54,7 @@ ISTAX_PROFILE_LIST =        \
     sparxIV_90_48           \
 
 # Custom/Stage2 := debug
+Custom/AddModules := demo_snmp
 
 # DefineTargetByPackage($1,$2)
 # $1: package name
diff --git a/vtss_appl/include/vtss/appl/module_id.h b/vtss_appl/include/vtss/appl/module_id.h
index 02620be..7bc4acab 100644
--- a/vtss_appl/include/vtss/appl/module_id.h
+++ b/vtss_appl/include/vtss/appl/module_id.h
@@ -212,6 +212,7 @@ enum {
     VTSS_MODULE_ID_FRER                 = 157,
     VTSS_MODULE_ID_KR                   = 158,
     VTSS_MODULE_ID_IEC_MRP              = 159,
+    VTSS_MODULE_ID_DEMO_SNMP            = 160,
 
     /*
      * INSERT NEW MODULE IDS HERE. AND ONLY HERE!!!
diff --git a/vtss_appl/main/main.cxx b/vtss_appl/main/main.cxx
index b2ec0df..d2b0fb5 100644
--- a/vtss_appl/main/main.cxx
+++ b/vtss_appl/main/main.cxx
@@ -476,6 +476,10 @@
 #include "vtss_optional_modules_api.h"
 #endif
 
+#if defined(VTSS_SW_OPTION_DEMO_SNMP)
+#include "demo_api.h"
+#endif
+
 #if defined(CYGPKG_IO_FILEIO) && defined(CYGPKG_FS_RAM) && defined(VTSS_SW_OPTION_ICFG)
 #include <unistd.h>
 #include <cyg/fileio/fileio.h>
@@ -902,6 +906,9 @@ static struct {
 #ifdef VTSS_SW_OPTION_PSFP
     INITFUN(psfp_init)
 #endif
+#ifdef VTSS_SW_OPTION_DEMO_SNMP
+    INITFUN(demo_init)
+#endif
 
 // **** NOTE: vtss_icfg_late_init must be last in this list ****
 #ifdef VTSS_SW_OPTION_ICFG
diff --git a/vtss_appl/util/vtss_module_id.cxx b/vtss_appl/util/vtss_module_id.cxx
index e74c260..dc53567 100644
--- a/vtss_appl/util/vtss_module_id.cxx
+++ b/vtss_appl/util/vtss_module_id.cxx
@@ -191,6 +191,7 @@ const char * const vtss_module_names[VTSS_MODULE_ID_NONE + 1] =
     [VTSS_MODULE_ID_FRER]                  /* 157 */ = "FRER",
     [VTSS_MODULE_ID_KR]                    /* 158 */ = "KR",
     [VTSS_MODULE_ID_IEC_MRP]               /* 159 */ = "IEC_MRP",
+    [VTSS_MODULE_ID_DEMO_SNMP]             /* 160 */ = "DEMO_SNMP",
 
     /* Add new module name above it. And please don't use space
        in module name, use underscore instead. */
@@ -541,6 +542,11 @@ const int vtss_priv_lvl_groups_filter[VTSS_MODULE_ID_NONE+1] =
     /*[VTSS_MODULE_ID_FRER]                  157 */ 1,
     /*[VTSS_MODULE_ID_KR]                    158 */ 1,
     /*[VTSS_MODULE_ID_IEC_MRP]               159 */ 1,
+#ifdef VTSS_SW_OPTION_DEMO_SNMP
+    /*[VTSS_MODULE_ID_DEMO_SNMP]             160 */ 0,
+#else
+    /*[VTSS_MODULE_ID_DEMO_SNMP]             160 */ 1,
+#endif
 
     /* Hint:
      * For a new module, if the module needs an independent privilege level group
